WooCommerce/Classes/ViewModels/ProductDetailsCellViewModel.swift

```diff
@@ -0,0 +1,143 @@
+import Foundation
+import Yosemite
+
+
+// MARK: - View Model for a product details cell
+//
+struct ProductDetailsCellViewModel {
+    // MARK: - Private properties
+
+    /// Yosemite.Order.currency
+    ///
+    private let currency: String
+
+    /// Yosemite.Product
+    ///
+    private let product: Product?
+
+    /// Currency Formatter
+    ///
+    private let currencyFormatter: CurrencyFormatter
+
+    /// Item Quantity
+    /// represented as a positive value
+    ///
+    private let positiveQuantity: Decimal
+
+    /// Item Total
+    /// represented as a positive value
+    ///
+    private let total: NSDecimalNumber
+
+    /// Item Price
+    /// represented as a positive value
+    ///
+    private let positivePrice: NSDecimalNumber
+
+    /// Item SKU
+    ///
+    private let skuText: String?
+
+    // MARK: - Public properties
+
+    /// Item Name
+    ///
+    var name: String
+
+    /// Item Quantity as a String
+    ///
+    var quantity: String {
+        return positiveQuantity.description
+    }
+
+    /// Refunded Product Price
+    /// Always return a string, even for zero amounts.
+    ///
+    var price: String {
+        guard positiveQuantity > 1 else {
+            return currencyFormatter.formatAmount(total, with: currency) ?? String()
+        }
+
+        let itemTotal = currencyFormatter.formatAmount(total, with: currency) ?? String()
+        let itemSubtotal = currencyFormatter.formatAmount(positivePrice, with: currency) ?? String()
+
+        let priceTemplate = NSLocalizedString("%@ (%@ x %@)",
+                                              comment: "<item total> (<item individual price> multipled by <quantity>)")
+        let priceText = String.localizedStringWithFormat(priceTemplate, itemTotal, itemSubtotal, quantity)
+
+        return priceText
+    }
+
+    /// Item SKU
+    ///
+    var sku: String? {
+        guard let sku = skuText, sku.isEmpty == false else {
+            return nil
+        }
+
+        let skuTemplate = NSLocalizedString("SKU: %@", comment: "SKU label, followed by the SKU")
+        let skuText = String.localizedStringWithFormat(skuTemplate, sku)
+
+        return skuText
+    }
+
+    /// Grab the first available image for a product.
+    ///
+    var imageURL: URL? {
+        guard let productImageURLString = product?.images.first?.src else {
+            return nil
+        }
+
+        return URL(string: productImageURLString)
+    }
+
+    // MARK: - Initializers
+
+    /// Order Item initializer
+    ///
+    init(item: OrderItem,
+         currency: String,
+         formatter: CurrencyFormatter = CurrencyFormatter(),
+         product: Product? = nil) {
+        self.currency = currency
+        self.currencyFormatter = formatter
+        self.product = product
+        self.name = item.name
+        self.positiveQuantity = abs(item.quantity)
```

Comment:
So the API reports all refunds as negative quantities, negative totals, and negative taxes. Because of the context of the screen (a list of all refunded products), it makes sense that the values are shown as positive (as the designs illustrate it). If you check the website, they are negative values because the page displays the math inline. I have preserved the negative values in the data models so that the math is correct when creating the aggregate order items.